# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/pulseaudio/hotfix-armv5.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- pulseaudio-16.1/meson.build.vanilla	2022-11-11 22:36:39.387133569 +0100
+++ pulseaudio-16.1/meson.build	2022-11-11 22:39:53.157150538 +0100
@@ -565,6 +565,7 @@
 if host_machine.cpu_family() == 'arm'
   armv6test = '''int func() {
     volatile int a = -60000, b = 0xaaaabbbb, c = 0xccccdddd;
+#if __ARM_ARCH > 5
     asm volatile ("ldr r0, %2 \n"
                   "ldr r2, %3 \n"
                   "ldr r3, %4 \n"
@@ -576,6 +577,9 @@
                   : "m" (a), "m" (b), "m" (c)
                   : "r0", "r1", "r2", "r3", "cc");
     return (a == -128 && b == 0xaabbdddd) ? 0 : -1;
+#else
+#error
+#endif
   }
   '''
 
